tuxemon/core/player.py

```diff
@@ -48,4 +48,17 @@ def __init__(self, npc_slug):
         self.isplayer = True
 
         # Game variables for use with events
-        self.game_variables = {}
+        self.game_variables = {
+        	'battle_history' : {} # Maps the names of NPCs the Player has battled to the outcome of their battle
+        						  #    any NPC name not present in the dictionary implies the Player has never 
+        						  #    battled them before
+        }
+
+    def set_state(self, game, save_data):

```

Comment:
good idea adding this check.  we have some save game migration code in https://github.com/Tuxemon/Tuxemon/blob/development/tuxemon/core/save_upgrader.py i would prefer that it be moved here.  i won't reject this pr though, i feel this is reasonable.  if you choose not to move it, please add a comment:
TODO: remove in future or move to save game migrations

Reply:
Thanks for finding this for me! I'll definitely move this check over so the code is cleaner.